coc/dataclasses.py

```diff
@@ -499,8 +498,8 @@ def __init__(self, *, data):
         self._data = data
         self.team_size = data.get('teamSize', None)
 
-        clan = data.get('clan', {})
-        if clan and ('tag' in clan.keys()):
+        clan = data.get('clan')
+        if clan and 'tag' in clan):
```

Comment:
nice. thought it would raise some attribute error since default is none but works nicely. 
Got an extra close bracket on line 502

Reply:
oops haha, will fix that and some more later, I think that tracking by token name may be the best, This would allow for multiple clients at once on different IPs